addons/web/static/src/views/list/list_renderer.js

```diff
@@ -239,6 +239,13 @@ export class ListRenderer extends Component {
         });
     }
 
+    async openFormView(record, newWindow) {
+        if (this.isX2Many && record.isNew) {
```

Comment:
I don't think its necessary, actually. See how is built the `openFormView` props in x2many_field.js. It already triggers the save. So maybe that only removing the condition in xml is enough. To check.